File: app/src/main/java/me/sunzheng/mana/LoginActivity.java
package me.sunzheng.mana;

import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import me.sunzheng.mana.account.AccountApiService;
import me.sunzheng.mana.account.AccountTableController;
import me.sunzheng.mana.account.LoginFragment;
import me.sunzheng.mana.account.config.HostFragment;
import me.sunzheng.mana.utils.App;

/**
 * A login screen that offers login via email/password.
 */
public class LoginActivity extends AppCompatActivity implements HostFragment.OnButtonClickListener, LoginFragment.OnFragmentInteractionListener {
    private final static String TAG = LoginActivity.class.getSimpleName();
    AccountTableController controller;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        controller = new AccountTableController(R.id.replace, getSupportFragmentManager());
    }


    @Override
    protected void onDestroy() {
        controller = null;
        super.onDestroy();
    }

    @Override
    public void onSave() {
        ((App) getApplicationContext()).initClient();
        controller.loginViewShow(((App) getApplicationContext()).getRetrofit().create(AccountApiService.Login.class));
    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }
}

File: app/src/main/java/me/sunzheng/mana/LoginActivity.kt
package me.sunzheng.mana

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.navigation.findNavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import dagger.hilt.android.AndroidEntryPoint
import me.sunzheng.mana.account.AccountViewModel
import me.sunzheng.mana.databinding.ActivityLoginBinding
import javax.inject.Inject

/**
 * A login screen that offers login via email/password.
 */
@AndroidEntryPoint
class LoginActivity @Inject constructor() : AppCompatActivity() {
    private lateinit var binding: ActivityLoginBinding
    private lateinit var appBarConfiguration: AppBarConfiguration

    val viewModel by viewModels<AccountViewModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_login)
        viewModel.isLoginLiveData.observe(this) {
            var id = when (it) {
                false -> {
                    R.navigation.nav_login
                }

                true -> {
                    R.navigation.nav_main
                }
            }
            var fragment = NavHostFragment.create(id)
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, fragment)
                .setPrimaryNavigationFragment(fragment)
                .commit()
        }
        replace()
    }

    fun replace() {
        viewModel.checkIsLogin(this)
    }

    override fun onNavigateUp(): Boolean {
        val navController = findNavController(supportFragmentManager.primaryNavigationFragment!!.id)
        return navController.navigateUp(appBarConfiguration)
                || super.onSupportNavigateUp()
    }

    companion object {
        private val TAG = LoginActivity::class.java.simpleName
    }
}